packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -231,3 +218,15 @@ export function evaluateTemplate(vm: VM, html: Template): Array<VNode | null> {
     }
     return vnodes;
 }
+
+export function hasScopedStyles(template: Template | null): boolean {
+    const stylesheets = template?.stylesheets;
+    if (!isUndefined(stylesheets) && stylesheets.length !== 0) {
+        for (let i = 0; i < stylesheets.length; i++) {
+            if (isTrue((stylesheets[i] as any)[KEY__SCOPED_CSS])) {
```

Comment:
I'm not super happy with putting an expando on a function as a way to figure out whether a stylesheet is scoped or not. Long term, I would prefer for this to just be a string, although it may have to be an object anyway, because exporting primitives in this case doesn't play well with `hot-swaps.ts`.